File: _2917_find_the_k_or_of_an_array/main.cpp
class Solution {
public:
    int findKOr(vector<int>& nums, int k) {
        int total = 0;
        int result = 0;

        for (int i = 0; i < 32; i++) {
            total = 0;
            for (const int &n : nums) {
                total += (n >> i) & 1;
            }

            if (total >= k) {
                result |= (1 << i);
            }
        }

        return result;
    }
};

File: _2917_find_the_k_or_of_an_array/main.py
class Solution:
    def findKOr(self, nums: List[int], k: int) -> int:
        result = 0

        for i in range(32):
            total = 0
            for n in nums:
                total += (n >> i) & 1

            if total >= k:
                result |= (1 << i)

        return result